# **INTRODUCTION**

Nama: Kamelia Ramandha.

Batch: CODA-RMT-010  

Live Coding 3 Phase 0 - Menggunakan pandas untuk data analysis




# Import library

In [78]:
import pandas as pd

In [79]:
df = pd.read_csv('netflix_movies.csv', delimiter = ',')

# Menjawab Pertanyaan

## Soal 1: Lakukan loading data

In [80]:
# Mengecek dataframe
df

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

## Soal 2: Tampilkan 5 data teratas

In [81]:
# Mengambil 5 data teratas menggunakan head()
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

## Soal 3: Tampilkan informasi struktur data. Bagaimana kondisi datanya? jawab menggunakan markdown!

In [82]:
# Melihat informasi struktur data menggunakan info()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


### Penjelasan data: data di atas memiliki 12 column dengan total data 8807. Tabel di atas menunjukan bahwa tidak ada nilai null di semua data, dengan data kebanyakan merupakan object dan hanya release_year yang merupakan string.

## Soal 4: Hapus kolom show_id

In [83]:
# Melakukan penghapusan kepada kolom show_id menggunakan del
del df['show_id']

df

type                  title         director  \
0       Movie   Dick Johnson Is Dead  Kirsten Johnson   
1     TV Show          Blood & Water              NaN   
2     TV Show              Ganglands  Julien Leclercq   
3     TV Show  Jailbirds New Orleans              NaN   
4     TV Show           Kota Factory              NaN   
...       ...                    ...              ...   
8802    Movie                 Zodiac    David Fincher   
8803  TV Show            Zombie Dumb              NaN   
8804    Movie             Zombieland  Ruben Fleischer   
8805    Movie                   Zoom     Peter Hewitt   
8806    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 11 columns]

## Soal 5: Ganti tipe data kolom date_added ke datetime64[ns]

In [84]:
# Membersihkan space kosong pada date_added(karena awalnya ada error karena adanya additional space)
df['date_added'] = df['date_added'].str.strip()

# Mengubah struktur date_Added menjadi datetime64
df['date_added'] = pd.to_datetime(df['date_added'], errors='coerce')

df

type                  title         director  \
0       Movie   Dick Johnson Is Dead  Kirsten Johnson   
1     TV Show          Blood & Water              NaN   
2     TV Show              Ganglands  Julien Leclercq   
3     TV Show  Jailbirds New Orleans              NaN   
4     TV Show           Kota Factory              NaN   
...       ...                    ...              ...   
8802    Movie                 Zodiac    David Fincher   
8803  TV Show            Zombie Dumb              NaN   
8804    Movie             Zombieland  Ruben Fleischer   
8805    Movie                   Zoom     Peter Hewitt   
8806    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

     date_added  release_year rating   duration  \
0    2021-09-25          2020  PG-13     90 min   
1    2021-09-24          2021  TV-MA  2 Seasons   
2    2021-09-24          2021  TV-MA   1 Season   
3    2021-09-24          2021  TV-MA   1 Season   
4    2021-09-24          2021  TV-MA  2 Seasons   
...         ...           ...    ...        ...   
8802 2019-11-20          2007      R    158 min   
8803 2019-07-01          2018  TV-Y7  2 Seasons   
8804 2019-11-01          2009      R     88 min   
8805 2020-01-11          2006     PG     88 min   
8806 2019-03-02          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 11 columns]

## Soal 6: Berapa banyak jumlah film (movies) dan TV Show pada data

In [85]:
# Melakukan perhitungan values untuk kolom type
df['type'].value_counts()

type
Movie      6131
TV Show    2676
Name: count, dtype: int64

## Soal 7: Film apa saja yang direlease di tahun 2021 yang berasal dari negara South Africa?

In [86]:
# Melakukan 2 filter terhadap data, yaitu release_year == 2021 dan country == South Africa
df[(df['release_year'] == 2021) & (df['country'] == 'South Africa') & (df['type'] == "Movie")]

type                         title       director  \
293   Movie                          Slay      Adze Ugah   
764   Movie  Trippin' with the Kandasamys  Jayan Moodley   
881   Movie                I Am All Girls  Donovan Marsh   
1172  Movie       Loyiso Gola: Unlearning  Kagiso Lediga   

                                                   cast       country  \
293   Enhle Mbali, Ramsey Nouah, Dawn Thandeka Kang,...  South Africa   
764   Jailoshini Naidoo, Maeshni Naicker, Madhushan ...  South Africa   
881   Erica Wessels, Hlubi Mboya, Deon Lotz, Brendon...  South Africa   
1172                                        Loyiso Gola  South Africa   

     date_added  release_year rating duration  \
293  2021-08-08          2021  TV-MA   86 min   
764  2021-06-04          2021  TV-14   94 min   
881  2021-05-14          2021  TV-MA  107 min   
1172 2021-03-23          2021  TV-MA   60 min   

                                            listed_in  \
293   Comedies, International Movies, Romantic Movies   
764   Comedies, International Movies, Romantic Movies   
881           Dramas, International Movies, Thrillers   
1172                                  Stand-Up Comedy   

                                            description  
293   In pursuit of both success and validation, a g...  
764   To rekindle their marriages, best friends-turn...  
881   A relentless detective finds common ground wit...  
1172  South African comedian Loyiso Gola serves up f...

## Soal 8 = Tampilkan 10 negara menghasilkan film terbanyak all time (urutkan dari yang terbanyak hingga paling sedikit)!

In [87]:
# Menampilkan data movie only dan menyusunnya berdasarkan dengan country. Lalu melakukan perhitungan kepada jumlah title yang ada dan mengurutkannya berdasarkan jumlah perhitungan title terbanyak dan menampilkan 10 data terbanyak
df[df['type'] == 'Movie'].groupby('country').agg(movie_count=('title','count')).sort_values('movie_count', ascending=False).head(10)

movie_count
country                    
United States          2058
India                   893
United Kingdom          206
Canada                  122
Spain                    97
Egypt                    92
Nigeria                  86
Indonesia                77
Japan                    76
Turkey                   76

## Soal 9 = Ambil data film yang bergenre Comedies

In [88]:
# Menyaring data listed_in yang memuat 'Comedies' dan type hanya 'Movies'
df[(df['listed_in'].str.contains('Comedies')) & (df['type'] == 'Movie')]

type                             title                 director  \
9     Movie                      The Starling           Theodore Melfi   
13    Movie  Confessions of an Invisible Girl            Bruno Garotti   
22    Movie                   Avvai Shanmughi           K.S. Ravikumar   
24    Movie                             Jeans               S. Shankar   
26    Movie                    Minsara Kanavu              Rajiv Menon   
...     ...                               ...                      ...   
8791  Movie                       Young Adult            Jason Reitman   
8793  Movie              Yours, Mine and Ours             Raja Gosnell   
8798  Movie                          Zed Plus  Chandra Prakash Dwivedi   
8804  Movie                        Zombieland          Ruben Fleischer   
8805  Movie                              Zoom             Peter Hewitt   

                                                   cast        country  \
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...  United States   
13    Klara Castanho, Lucca Picon, Júlia Gomes, Marc...            NaN   
22    Kamal Hassan, Meena, Gemini Ganesan, Heera Raj...            NaN   
24    Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...          India   
26    Arvind Swamy, Kajol, Prabhu Deva, Nassar, S.P....            NaN   
...                                                 ...            ...   
8791  Charlize Theron, Patton Oswalt, Patrick Wilson...  United States   
8793  Dennis Quaid, Rene Russo, Sean Faris, Katija P...  United States   
8798  Adil Hussain, Mona Singh, K.K. Raina, Sanjay M...          India   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   

     date_added  release_year rating duration  \
9    2021-09-24          2021  PG-13  104 min   
13   2021-09-22          2021  TV-PG   91 min   
22   2021-09-21          1996  TV-PG  161 min   
24   2021-09-21          1998  TV-14  166 min   
26   2021-09-21          1997  TV-PG  147 min   
...         ...           ...    ...      ...   
8791 2019-11-20          2011      R   94 min   
8793 2019-11-20          2005     PG   88 min   
8798 2019-12-31          2014  TV-MA  131 min   
8804 2019-11-01          2009      R   88 min   
8805 2020-01-11          2006     PG   88 min   

                                             listed_in  \
9                                     Comedies, Dramas   
13                  Children & Family Movies, Comedies   
22                      Comedies, International Movies   
24     Comedies, International Movies, Romantic Movies   
26    Comedies, International Movies, Music & Musicals   
...                                                ...   
8791              Comedies, Dramas, Independent Movies   
8793                Children & Family Movies, Comedies   
8798            Comedies, Dramas, International Movies   
8804                           Comedies, Horror Movies   
8805                Children & Family Movies, Comedies   

                                            description  
9     A woman adjusting to life after a loss contend...  
13    When the clever but socially-awkward Tetê join...  
22    Newly divorced and denied visitation rights wi...  
24    When the father of the man she loves insists t...  
26    A tangled love triangle ensues when a man fall...  
...                                                 ...  
8791  When a divorced writer gets a letter from an o...  
8793  When a father of eight and a mother of 10 prep...  
8798  A philandering small-town mechanic's political...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  

[1674 rows x 11 columns]

## Soal 10: Tampilkan film-film yang mengandung kata land di judulnya! (bisa gunakan method .str.contains)

In [89]:
# Menyaring title yang mengandung 'land' dan menyaring type hanya 'Movie'
df[(df['title'].str.contains('land')) & (df['type'] == 'Movie')]

type                                              title  \
53    Movie    InuYasha the Movie 4: Fire on the Mystic Island   
252   Movie                                       Black Island   
335   Movie                                        Freedomland   
939   Movie                           Motu Patlu in Wonderland   
1222  Movie                             The Block Island Sound   
1358  Movie                                     Shutter Island   
1571  Movie                                        Rose Island   
1604  Movie                   Ari Eldjárn: Pardon My Icelandic   
1719  Movie                      We Are All For The Fatherland   
1816  Movie                           Tremors: Shrieker Island   
2119  Movie                                   Islands of Faith   
2215  Movie                             The Larva Island Movie   
2262  Movie                            A Kid from Coney Island   
3166  Movie                                Banana Island Ghost   
3217  Movie                                         The Island   
3331  Movie                         Christmas in the Heartland   
3586  Movie                             The Little Switzerland   
3802  Movie  The Lonely Island Presents: The Unauthorized B...   
3921  Movie           Who Would You Take to a Deserted Island?   
3967  Movie                       The Legend of Cocaine Island   
4222  Movie                      Moshe Kasher: Live in Oakland   
4603  Movie                             Sat Shri Akaal England   
5093  Movie                                     Mustang Island   
5142  Movie                                     Monster Island   
5196  Movie  Judah Friedlander: America Is the Greatest Cou...   
5271  Movie                                      Strong Island   
5624  Movie                               Journey to Greenland   
6140  Movie             American Experience: The Island Murder   
6154  Movie  An American Tail: The Treasures of Manhattan I...   
6218  Movie                                            Badland   
6340  Movie          Blue Mountain State: The Rise of Thadland   
6356  Movie                                   Bon Bini Holland   
6989  Movie                                           Homeland   
7260  Movie                                          Laddaland   
7276  Movie                               LeapFrog: Numberland   
7959  Movie                        Scooby-Doo on Zombie Island   
7975  Movie                           Secrets of Scotland Yard   
8075  Movie              Spy Kids 2: The Island of Lost Dreams   
8530  Movie                             The Tigers of Scotland   
8733  Movie                                       White Island   
8804  Movie                                         Zombieland   

                                             director  \
53                                  Toshiya Shinohara   
252                                  Miguel Alexandre   
335                                          Joe Roth   
939                                       Suhas Kadav   
1222                   Kevin McManus, Matthew McManus   
1358                                  Martin Scorsese   
1571                                   Sydney Sibilia   
1604                                 August Jakobsson   
1719                                              NaN   
1816                                 Don Michael Paul   
2119                                    Chairun Nissa   
2215                                  Ahn Byoung-wook   
2262                                    Coodie, Chike   
3166                                        BB Sasore   
3217                                     Toka McBaror   
3331                                     Harvey Lowry   
3586                                        Kepa Sojo   
3802                        Mike Diva, Akiva Schaffer   
3921                                     Jota Linares   
3967                                        Theo Love   
4222                                 

## Soal 11 = Siapa director yang paling banyak menghasilkan film dan TV Show?

In [70]:
# Menyusun data berdasarkan director, menghitung jumlah title dan mengurutkan berdasarkan title terbanyak. Hanya menampilkan 1 nama teratas
df.groupby('director').agg(show_count=('title','count')).sort_values('show_count', ascending=False).head(1)

show_count
director                 
Rajiv Chilaka          19

## Soal 12 = Negara mana saja yang menghasilkan TV Show terbanyak?

In [90]:
# Memfilter type hanya TV Show dan mengkategorikannya berdasarkan country. Melakukan perhitungan jumlah title dan mengurutkannya secara descending. Hanya menampilkan 1 data teratas
df[df['type'] == 'TV Show'].groupby('country').agg(tvshow_count=('title','count')).sort_values('tvshow_count', ascending=False).head(1)

tvshow_count
country                    
United States           760